### PR TITLE
Add resets to heading typography variants

### PR DESCRIPTION
**Pull request by nickbrowne**
This isn't ideal, but it was far harder than expected to quarantine TC's styles. (biggest blocker and bordering on impossible to change at the moment is the content pipeline). This ended up being the easiest way to prevent TC's legacy CSS from messing with MUI typography heading variants.

Initially I planned to reset every property that we set in TC (like the trello card mentions), but that caused additional problems due to class specificity and the way MUI mixes classes together to build the styles for Typography components.

In the end, I experimented until I only had resets for just the properties that were likely to cause (and have caused) the most grief while effectively being a no-op visually - so if you are testing locally, you shouldn't see any visual differences in this PR.

This may mean we could encounter problems in less travelled parts of TC, but we can always add more resets if needed.

### CODE REVIEW

**Comment by markcipolla** [comment]
@nickbrowne So, to test this, would this be `npm link`?

**Comment by markcipolla** [comment]
Are there any places on TC you've seen where it would fix problems?

**Comment by nickbrowne** [comment]
> @nickbrowne So, to test this, would this be `npm link`?

If that works for you, I just did an `npm install` with the absolute path.

> Are there any places on TC you've seen where it would fix problems?

The most obvious ones have already been fixed, but I couldn't continue with that way of fixing things because it became too difficult:

https://github.com/conversation/tc/pull/9275
https://github.com/conversation/tc/pull/9313
https://github.com/conversation/tc/pull/9337

I will likely revert some of those changes after this merges and we cut a new version.

**Comment by jameshill** [comment]
@nickbrowne oh interesting so you're going to revert those PRs and remove the legacy class?

That's a shame.. what ended up being the straw that broke the camels back in content pipeline?

I have no idea the best way to isolate those components, but you couldn't target them by a class on the content body's container... 

we wouldn't be able to have tc/ui inside the content body though, but we're not do that at the moment.

**Comment by nickbrowne** [comment]
> That's a shame.. what ended up being the straw that broke the camels back in content pipeline?

Without going into too much detail, Rohan and I tried to extend the markdown library (according to a blog post from 2013 that only exists on the internet archive) to have more control over how it outputs heading elements. 

But because it's c extensions and metaprogramming, it was far harder than expected. Then when we finally got it working, we ran into problems where we ended up with those classes ended up in unexpected places like feeds and apple news.

We thought about doing some gsubbing, but then I backed away from it all together, I wasn't willing to add more complexity and switches to the content pipeline.

We also tried the more targeted css that you mentioned, but that broke a bunch of other styles because the change in specificity meant that the way the styles cascaded was different. There's also the issue that it would break ui components rendered into a content body even worse, like you mentioned.

Basically all roads led to hell :fire:, so I think the best bet in the short term is what we've got here and then visually unifying the two worlds when we're ready.

**Comment by jameshill** [comment]
Sounds good... I can't test this at the moment, but it gets a 👍 from me... if someone else can have a play with it alongside TC that would be marvellous.

**Comment by markcipolla** [comment]
> If that works for you, I just did an `npm install` with the absolute path.

I've tried `npm install` and `npm link`, and no luck; `npm i` with an absolute path won't build, and `npm link` seems to do nothing (or, more likely, I'm using it wrong).

@nickbrowne Are you able to post up some steps on how to manually test this?

**Comment by markcipolla** [comment]
@nickbrowne did a screen share (the code looked good, but nice to see it in action)

👍

**Comment by nickbrowne** [comment]
> @nickbrowne Are you able to post up some steps on how to manually test this?

Will open a PR to add something to the readme because I also wasted a bunch of time trying to get it to work again today.
